File: core/core-domain/src/main/java/com/nmh/commerce/exception/DomainException.java
package com.nmh.commerce.exception;

import com.nmh.commerce.support.error.CustomException;
import com.nmh.commerce.support.error.ErrorType;

public class DomainException extends CustomException {
    public DomainException(ErrorType errorType) {
        super(errorType);
    }

    public DomainException(ErrorType errorType, Object data) {
        super(errorType, data);
    }
}

File: core/core-domain/src/main/kotlin/com/nmh/commerce/exception/DomainException.kt
package com.nmh.commerce.exception

import com.nmh.commerce.support.error.CustomException
import com.nmh.commerce.support.error.ErrorType

class DomainException : CustomException {
    constructor(errorType: ErrorType) : super(errorType)

    constructor(errorType: ErrorType, data: Any?) : super(errorType, data)
}
